packages/flutter/lib/src/foundation/_platform_web.dart

```diff
@@ -19,46 +18,45 @@ platform.TargetPlatform get defaultTargetPlatform {
       _browserPlatform;
 }
 
-final platform.TargetPlatform? _testPlatform = () {
-  platform.TargetPlatform? result;
+// The TargetPlatform used ON WEB TESTS, unless overridden.
+//
+// Respects the `ui_web.browser.debugOperatingSystemOverride` value (when set).
+platform.TargetPlatform? get _testPlatform {
+  platform.TargetPlatform? testPlatform;
```

Comment:
Note that this used to be a `final` variable computed from an IIFE, but it needs to be a `getter` so it can be overridden as needed from tests. The getter ends up as an empty function when asserts are disabled, so I hope this doesn't cause a huge performance hit.

Reply:
Maybe `@pragma('dart2js:tryInline')` could help?

Reply:
Brought this up with the dart2js people, and they confirmed that the annotation is not needed; @biggs0125 said:

> Dart2js should do a pretty good job of inlining that code when assertions are disabled, even without the annotation. We should be able to tell that (1) the code has no side effects and (2) the code always returns null. So it should just inline a null into the callsite.

(I hope it doesn't make `debug` mode too miserable! In that case, I'd walk back the changes that I added just for testing, and find another way)